junit-platform-console/src/main/java/org/junit/platform/console/tasks/DiscoveryRequestCreator.java

```diff
@@ -127,4 +129,14 @@ private void addFilters(LauncherDiscoveryRequestBuilder requestBuilder, CommandL
 		}
 	}
 
+	private ClassNameFilter includedClassNamePatterns(CommandLineOptions patterns) {
+		List<String> classNamePatterns = new ArrayList<>();
+		classNamePatterns.addAll(patterns.getIncludedClassNamePatterns());
+		classNamePatterns.addAll(patterns.getSelectedClasses());
+		classNamePatterns.addAll(patterns.getSelectedMethods().stream() //
+				.map(name -> ReflectionUtils.parseFullyQualifiedMethodName(name)[0]) //
+				.collect(Collectors.toList()));
+		return includeClassNamePatterns(classNamePatterns.toArray(new String[0]));
+	}
```

Comment:
We should quote the literal class names using `Pattern.quote`, e.g. like this:

```suggestion
	private ClassNameFilter includedClassNamePatterns(CommandLineOptions options) {
		Stream<Stream<String>> patternStreams = Stream.of( //
			options.getIncludedClassNamePatterns().stream(), //
			options.getSelectedClasses().stream() //
					.map(Pattern::quote), //
			options.getSelectedMethods().stream() //
					.map(name -> ReflectionUtils.parseFullyQualifiedMethodName(name)[0]) //
					.map(Pattern::quote));
		return includeClassNamePatterns(patternStreams //
				.flatMap(Function.identity()) //
				.toArray(String[]::new));
	}
```

Reply:
Haven't thought of that, thanks! 👍